agent/consul/state/catalog_events.go

```diff
@@ -0,0 +1,514 @@
+package state
+
+import (
+	"github.com/hashicorp/consul/agent/agentpb"
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/consul/agent/structs"
+	memdb "github.com/hashicorp/go-memdb"
+)
+
+// _events.go files contain the streaming event methods relevant to specific
+// state store components.
+
+// ServiceHealthSnapshot is a stream.SnapFn that provides a streaming snapshot
+// of agentpb.Events that describe the current state of a service health query.
+func (s *Store) ServiceHealthSnapshot(req *agentpb.SubscribeRequest, buf *stream.EventBuffer) (uint64, error) {
+	tx := s.db.Txn(false)
+	defer tx.Abort()
+	// TODO(banks): plumb entMeta through from SubscribeRequest
```

Comment:
suggestion: I've found it slightly nicer to use topical TODOs instead of personal ones, like `TODO(namespaces)` or `TODO(streaming)`.